scripts/upgrade-hooks/1.1.0.ts

```diff
@@ -0,0 +1,251 @@
+  }
+
+  const url = subgraphConfig[env][chainId.toString()].subgraphUrl;
+  if (url === "TBD") {
+    throw new Error(`Subgraph URL is TBD for chainId ${chainId} and env ${env}`);
+  }
+
+  return url;
+}
+
+// Function to get the InitializationFacet address based on chainId and env
+function getInitializationFacetAddress(chainId: number, env: string): string {
+  const addressesDir = path.join(__dirname, "..", "..", "addresses");
+  const addressFile = fs.readFileSync(path.join(addressesDir, `${chainId}-${env}.json`), "utf8");
+  const addressData = JSON.parse(addressFile);
+
+  const initializationFacet = addressData.contracts.find((contract: any) => contract.name === "InitializationFacet");
+
+  if (!initializationFacet) {
+    throw new Error(`InitializationFacet not found in addresses file for chainId ${chainId} and env ${env}`);
+  }
+
+  return initializationFacet.address;
+}
+
+interface Token {
+  id: string;
+  status: number;
+  priceLog: { amount: string | null };
+}
+
+interface Offer {
+  verifierFee: string;
+  facilitatorFeePercent: string;
+  fNFTs: Token[];
+}
+
+export interface FeeData {
+  tokenId: string;
+  bosonProtocolFee: bigint;
+  fermionFeeAmount: bigint;
+  verifierFee: bigint;
+  facilitatorFeeAmount: bigint;
+}
+
+const OFFER_QUERY = `
+  query {
+    offers(where: { fNFTs_: { status_in: [3, 4, 5, 6, 7] } }) {
+      verifierFee
+      facilitatorFeePercent
+      fNFTs {
+        id
+        status
+        priceLog {
+          amount
+        }
+      }
+    }
+  }
+`;
+
+const FNFT_RANGE_QUERY = `
+  query {
+    fnftranges {
+      bosonOfferId
+      startingId
+      quantity
+    }
+  }
+`;
+
+interface FNFTRange {
+  bosonOfferId: string;
+  startingId: string;
+  quantity: string;
+}
+
+export interface OfferData {
+  offerId: string;
+  itemQuantity: string;
+  firstTokenId: string;
+}
+
+/**
+ * Fetch GraphQL data for FNFT ranges.
+ */
+async function fetchFNFTRangeData(client: any): Promise<FNFTRange[]> {
+  const result = await client.query(FNFT_RANGE_QUERY, {}).toPromise();
+
+  if (result.error) {
+    throw new Error(`GraphQL error: ${result.error.message}`);
+  }
+
+  if (!result.data?.fnftranges) {
+    throw new Error("No FNFT range data found in GraphQL response");
+  }
+
+  return result.data.fnftranges;
+}
+
+/**
+ * Prepare backfill data for offers.
+ */
+export async function prepareOfferBackfillData(client: any): Promise<OfferData[]> {
+  const fnftRanges = await fetchFNFTRangeData(client);
+  const offerDataList: OfferData[] = [];
+
+  for (const range of fnftRanges) {
+    offerDataList.push({
+      offerId: range.bosonOfferId,
+      itemQuantity: range.quantity,
+      firstTokenId: range.startingId,
+    });
+  }
+
+  console.log("Prepared Offer Backfill Data:", JSON.stringify(offerDataList, null, 2));
+
+  return offerDataList;
+}
+
+/**
+ * Fetch GraphQL data for offers and tokens in specific states.
+ */
+async function fetchGraphQLData(client: any): Promise<Offer[]> {
```

Comment:
Maybe a bit misleading name if we have `fetchFNFTRangeData` for the other case, we could have `fetchOfferData` or sth similar here.